### ISSUE TITLE
Implement Curved Bond Artonomous Soul Tokens

### ISSUE THREAD

**Issue by simondlr** [simondlr/artonomous#4]
In order to give the autonomous artist life, benefactors can purchase soul tokens based on a curved bond. https://medium.com/@simondlr/tokens-2-0-curved-token-bonding-in-curation-markets-1764a2e0bee5.

- The soul tokens serve as a claim on the bonded ETH to keep the artonomous alive. 
- Any art sold through the auction contract gets deposited into the bond.
- The entry curved should be an exponential curve.
- The exit curve is percentage-based. It allows a claim on all outstanding ETH in the bond proportional to the amount you are removing from supply.
- The soul tokens are used to stake on the hash of the software currently generating new art.

**Comment by simondlr** [comment]
https://github.com/oed/bonding-curves

**Comment by simondlr** [comment]
Ah, I see what you mean. The ETH changing the curve price would be helpful.

I was thinking that the outbound curve (sell) was just percentage based. If there's 100 ETH in the pool and 100 soul tokens outstanding and you sell 1 soul token, you 1% of the pot.

However, if it doesn't affect the curve (what my intention was), then it essentially means that will kind of awkwardly rebalance. It will profitable to buy and then immediately sell a soul token.

I like it better if the curve is directly affect only based on the ETH in the pool. It's better that way. So it seems a BANCOR curve would be better in this scenario. +1

**Comment by simondlr** [comment]
> IMO the actions of minting SOUL and burning it immediately after should cancel out.

I agree, yes.

There are however scenarios where no minting would result in getting a higher token price if the reserve grows from an art sale.

If ```PRICE = ETH_POOL/(SOUL_SUPPLY*RATIO)``` then if an art sale comes in, everyone's tokens increases in value.

**Comment by mzargham** [comment]
Agree, the cancel out assumes no intermediate state change. We can define the requirement with regard to “immediately” = “no intermediate state change”

Of course if for any reason the system state evolves due any process other than a Bond or Withdraw Ether (which both respect this curve), then they will not just cancel out. I think this means we’ll have the same linear bonding curve on both the Bond and Withdraw mechanisms. Makes intuitive sense to me and counters the argument that its a Ponzi because all net profits from bonding and withdrawing later come from the art auction process not from preying on later investors.

**Comment by BenSchZA** [comment]
@mzargham and others ~ you might find this thread and tool useful for LaTeX type math formatting: https://github.com/github/markup/issues/897 & https://www.codecogs.com/latex/eqneditor.php

Not officially supported, but does make the math easier to follow :slightly_smiling_face: 

For example, taking equation above:

<a href="https://www.codecogs.com/eqnedit.php?latex=E_&plus;/S_&plus;&space;=&space;\frac{E-e}{S-s}&space;=&space;\frac{E-\frac{sE}{S}}{S-s}&space;=&space;E\frac{1-s}{S}&space;/&space;S\frac{1-s}{S}&space;=&space;E/S" target="_blank"><img src="https://latex.codecogs.com/gif.latex?E_&plus;/S_&plus;&space;=&space;\frac{E-e}{S-s}&space;=&space;\frac{E-\frac{sE}{S}}{S-s}&space;=&space;E\frac{1-s}{S}&space;/&space;S\frac{1-s}{S}&space;=&space;E/S" title="E_+/S_+ = \frac{E-e}{S-s} = \frac{E-\frac{sE}{S}}{S-s} = E\frac{1-s}{S} / S\frac{1-s}{S} = E/S" /></a>
